afs/afs-core/src/main/java/com/powsybl/afs/ProjectFile.java

```diff
@@ -16,11 +20,33 @@
 
     protected final FileIcon icon;
 
-    private final WeakHashMap<Object, List<DependencyListener>> listeners = new WeakHashMap<>();
+    private final WeakHashMap<Object, List<ProjectFileListener>> listeners = new WeakHashMap<>();
 
     protected ProjectFile(ProjectFileCreationContext context, int codeVersion, FileIcon icon) {
         super(context, codeVersion, true);
         this.icon = Objects.requireNonNull(icon);
+        storage.addListener(this, eventList -> {
+            for (NodeEvent event : eventList.getEvents()) {
+                switch (event.getType()) {
+                    case DEPENDENCY_ADDED:
+                    case DEPENDENCY_REMOVED:
+                        listeners.values().stream()
```

Comment:
Do we need to copy the listeners in a new list as it's done in AppStorageListenerList.java?